python/modules/BtagSFWeightProducer.py

```diff
@@ -175,20 +197,32 @@ def analyze(self, event):
                 BTagWeightN      *= 1 - btagSF * eff
                 BTagWeightN_up   *= 1 - btagSF_up * eff
                 BTagWeightN_down *= 1 - btagSF_down * eff
-
-		if abs(flavor) == 5:
-                	BTagWeightNHeavy      *= 1 - btagSF * eff
-                	BTagWeightNHeavy_up   *= 1 - btagSF_up * eff
-                	BTagWeightNHeavy_down *= 1 - btagSF_down * eff
-			BTagWeightDHeavy      *= 1 - eff
-		else:
-                	BTagWeightNLight      *= 1 - btagSF * eff
-                	BTagWeightNLight_up   *= 1 - btagSF_up * eff
-                	BTagWeightNLight_down *= 1 - btagSF_down * eff
-			BTagWeightDLight      *= 1 - eff
+                BTagWeightN_up_FS   *= 1 - btagSF_up_FS * eff
+                BTagWeightN_down_FS *= 1 - btagSF_down_FS * eff
+
+                if abs(flavor) == 5:
+                        BTagWeightNHeavy      *= 1 - btagSF * eff
+                        BTagWeightNHeavy_up   *= 1 - btagSF_up * eff
+                        BTagWeightNHeavy_up_FS   *= 1 - btagSF_up_FS * eff
+                        BTagWeightNHeavy_down_FS *= 1 - btagSF_down_FS * eff
+                        BTagWeightDHeavy      *= 1 - eff
+                else:
+                    BTagWeightNLight       *= 1 - btagSF * eff
+                    BTagWeightNLight_up    *= 1 - btagSF_up * eff
+                    BTagWeightNLight_up_FS *= 1 - btagSF_up_FS * eff
+                    BTagWeightNLight_down  *= 1 - btagSF_down * eff
+                    BTagWeightNLight_down_FS  *= 1 - btagSF_down_FS * eff
+                    BTagWeightDLight      *= 1 - eff
 
                 BTagWeightD      *= 1 - eff
-
+         
+        if self.FastSim:
+            self.out.fillBranch("BTagWeight_Up_FS",   BTagWeightN_up_FS / BTagWeightD)
```

Comment:
How do we apply this FS uncertainty, if all as a ratio to BTagWeightD?

Reply:
you want the ratio of events with "BTagWeight_Up_FS " / "BTagWeight"

The denominator is only a function of the efficiencies, so no SF (fast or fullsim) will affect this.

Reply:
Ha,  BTagWeightD *= eff, but BTagWeightN  *= btagSF * eff *= jet.btagSF*jet.btagSF_F*eff. I got these two mixed up.